tests/span/test_base_span.py

```diff
@@ -1,158 +1,182 @@
-    assert _kwarg1 == base_span.arg1
-
-
-def test_populate_extra_span_attributes(span: InstanaSpan) -> None:
-    base_span = BaseSpan(span, None)
-    base_span._populate_extra_span_attributes(span)
-
-    assert not hasattr(base_span, "tp")
-    assert not hasattr(base_span, "tp")
-    assert not hasattr(base_span, "ia")
-    assert not hasattr(base_span, "lt")
-    assert not hasattr(base_span, "crtp")
-    assert not hasattr(base_span, "crid")
-
-
-def test_populate_extra_span_attributes_with_values(
-    trace_id: int,
-    span_id: int,
-    span_processor: StanRecorder,
-) -> None:
-    long_id = 1512366075204170929049582354406559215
-    span_context = SpanContext(
-        trace_id=trace_id,
-        span_id=span_id,
-        is_remote=False,
-        synthetic=True,
-        trace_parent=True,
-        instana_ancestor="IDK",
-        long_trace_id=long_id,
-        correlation_type="IDK",
-        correlation_id=long_id,
-    )
-    span = InstanaSpan("test-base-span", span_context, span_processor)
-    base_span = BaseSpan(span, None)
-    base_span._populate_extra_span_attributes(span)
-
-    assert trace_id == base_span.t
-    assert span_id == base_span.s
-    assert base_span.sy
-    assert base_span.tp
-    assert "IDK" == base_span.ia
-    assert long_id == base_span.lt
-    assert "IDK" == base_span.crtp
-    assert long_id == base_span.crid
-
-
-def test_validate_attributes(base_span: BaseSpan) -> None:
-    attributes = {
-        "field1": 1,
-        "field2": "two",
-    }
-    filtered_attributes = base_span._validate_attributes(attributes)
-
-    assert isinstance(filtered_attributes, dict)
-    assert len(attributes) == len(filtered_attributes)
-    for key, value in attributes.items():
-        assert key in filtered_attributes.keys()
-        assert value in filtered_attributes.values()
-
-
-def test_validate_attribute_with_invalid_key_type(base_span: BaseSpan) -> None:
-    key = 1
-    value = "one"
-
-    (validated_key, validated_value) = base_span._validate_attribute(key, value)
-
-    assert not validated_key
-    assert not validated_value
-
-
-def test_validate_attribute_exception(span: InstanaSpan) -> None:
-    base_span = BaseSpan(span, None)
-    key = "field1"
-    value = span
-
-    with patch(
-        "instana.span.base_span.BaseSpan._convert_attribute_value",
-        side_effect=Exception("mocked error"),
-    ):
-        (validated_key, validated_value) = base_span._validate_attribute(key, value)
-        assert key == validated_key
-        assert not validated_value
-
-
-def test_convert_attribute_value(span: InstanaSpan) -> None:
-    base_span = BaseSpan(span, None)
-    value = span
-
-    converted_value = base_span._convert_attribute_value(value)
-    assert "<instana.span.span.InstanaSpan object" in converted_value
+class TestBaseSpan:
+    @pytest.fixture(autouse=True)
+    def _resource(self) -> Generator[None, None, None]:
+        self.base_span = None
+        yield
```

Comment:
If this is the only change in this file, I don't see any reason to create a class for the whole test. One of the advantages of using pytest is that it is unnecessary to create a test class to run tests.

Reply:
Isn't having the same structure in all test files a good behaviour?

Reply:
IMHO, only when necessary. That's not the case here.